core/src/main/java/net/opentsdb/query/processor/groupby/GroupByNumericArrayIterator.java

```diff
@@ -128,17 +152,38 @@ public GroupByNumericArrayIterator(final QueryNode node,
     }
     
     long startTs = DateTime.nanoTime();
+    
+    TemporalAmount temporalAmount = this.result.downstreamResult().timeSpecification().interval();
+    
+    TimeStamp ts = node.pipelineContext().query().startTime().getCopy();
+    int intervals = 0;
+    while (ts.compare(Op.LT, node.pipelineContext().query().endTime())) {
+      intervals++;
+      ts.add(temporalAmount);
+    }
+    
+    logger.info("Sources size {}, Intervals found {}", sources.size(), intervals);
+
+    // interval * 2 to capture sum and count; [sum1,count1 ... sumn,countn]
+    valuesCombiner = new double[NUM_THREADS][intervals * 2];
+
+    // init with NaN
+    for (int i = 0; i < NUM_THREADS; i++) {
+      Arrays.fill(valuesCombiner[i], Double.NaN);
+    }
 
     // TODO: Need to check if it makes sense to make this threshold configurable
-    if (sources.size() >= 1000) {
+    if (sources.size() >= NUM_THREADS) {
```

Comment:
the `num_thereads` should be configurable with a default value of 8.

Reply:
Yeah should be configurable but at this spot in the code it's the # of time series we want to switch to parallizing.